src/syscheckd/registry/registry.c

```diff
@@ -756,23 +726,178 @@ void fim_read_values(HKEY key_handle,
         DWORD data_size = max_value_data_length;
         DWORD data_type = 0;
 
+        configuration = fim_registry_configuration(path, arch);
+        if (configuration == NULL) {
+            return;
+        }
+
         if (RegEnumValue(key_handle, i, value_buffer, &value_size, NULL, &data_type, data_buffer, &data_size) !=
             ERROR_SUCCESS) {
             break;
         }
 
-        new->registry_entry.value->name = value_buffer;
-        new->registry_entry.value->type = data_type;
-        new->registry_entry.value->size = data_size;
-        new->registry_entry.value->last_event = time(NULL);
+        new.registry_entry.value->name = value_buffer;
+        new.registry_entry.value->type = data_type;
+        new.registry_entry.value->size = data_size;
+        new.registry_entry.value->last_event = 0;
+        new.registry_entry.value->scanned = 0;
+        new.type = FIM_TYPE_REGISTRY; //arreglado uninitialized read
```

Comment:
Could you write this comment in English?

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/11239/commits/11a6215594b89425c4469067a26e23d1a1a7f5df)